File: practica3/python/pila.py
class Pila:
    def __init__(self, n):
        self.arreglo = [None] * n  # Lista con tamaño fijo
        self.top = -1
        self.n = n

    def push(self, e):
        if self.is_full():
            print("Pila llena")
        else:
            self.top += 1
            self.arreglo[self.top] = e

    def pop(self):
        if self.is_empty():
            print("Pila vacía")
            return -1
        else:
            e = self.arreglo[self.top]
            self.top -= 1
            return e

    def peek(self):
        if self.is_empty():
            print("Pila vacía")
            return -1
        return self.arreglo[self.top]

    def is_empty(self):
        return self.top == -1

    def is_full(self):
        return self.top == self.n - 1


# Prueba de la implementación
if __name__ == "__main__":
    pila = Pila(3)
    pila.push(10)
    pila.push(20)
    pila.push(30)

    print(pila.peek())
    print("pop() =", pila.pop())
    print("pop() =", pila.pop())
    print("pop() =", pila.pop())

File: practica3/Java/Pila.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pila;

/**
 *
 * @author Administrador
 */
public class Pila {

   public int arreglo[];
   public int top;
   public int n;
   public Pila(int n){
       arreglo =new int[n];
       top=-1;
       this.n=n;
   }
   public void push(int e){
       if(isFull()){
           System.out.println("Pila llena");
       }else{
           top++;
           arreglo[top]=e;
       }
   }
   public int pop(){
       if(isEmpty()){
          System.out.println("Pila vacia");
          return -1;
       }else{
           int e=arreglo[top];
           top--;
           return e;
       }
   }
   public int peek(){
      return arreglo[top];
   }
   
    private boolean isEmpty() {
        return (top ==-1);
    }
    private boolean isFull(){
        return (top==n-1);
    }
    public static void main(String[] args) {
        Pila pila =new Pila(3);
        pila.push(10);
        pila.push(20);
        pila.push(30);
        
        System.out.println(pila.peek());
        System.out.println("pop() = "+pila.pop());
        System.out.println("pop() = "+pila.pop());
        System.out.println("pop() = "+pila.pop());
       
           
    }

  
    
}
